server/aesdsocket.c

```diff
@@ -203,9 +218,26 @@ static int read_packet(int socket, int fd, pthread_mutex_t *m) {
         }
     }
 
-    // write data to file /var/tmp/aesdsocketdata
+    #ifdef USE_AESD_CHAR_DEVICE
+        // Check to see if we've gotten the ioctl command and the other arguments
+        ioctl_found = strncmp(buff, IOCTL_CMD_STR, IOCTL_STR_CMD_LEN);
+        if(ioctl_found == 0) {
+            if(byte_read >= MINIMUM_IOCTL_CMD_LEN) {
+                syslog(LOG_INFO, "IOCTL command received");
+                
+                // call the ioctl method and skip writting to file
+                seekto.write_cmd        = (uint32_t)atoi(&buff[IOCTL_STR_CMD_LEN]);
+                seekto.write_cmd_offset = (uint32_t)atoi(&buff[MINIMUM_IOCTL_CMD_LEN - 1]);
```

Comment:
@KapureCUB I think this logic won't work if the write_cmd is > 10, since the buffer offset will be wrong.